### ISSUE TITLE
Move RoutablePageMixin into wagtailcore?

### ISSUE THREAD

**Issue by kaedroho** [torchbox/wagtail#1012]
Seems like a more logical place for it I think. It's incredably useful. Maybe we could also merge ``routablepageurl`` with ``pageurl``.

**Comment by gasman** [comment]
Not keen on the idea, I'm afraid!

Being incredibly useful is not reason enough - django.contrib.admin is also incredibly useful, but that hasn't been promoted to Django core :-)

I think it's healthy for the modules to be separated, to keep the core Page API focused and un-bloated. Ideally, someone should be able to read wagtailcore.models in its entirety, and have it entirely make sense for the intuitive concept of a "page" as a thing with a URL and an HTML representation. The fact that you can then take that core API, and build useful and interesting things like RoutablePage and wagtailforms on top of it - even though there's nothing in wagtailcore that's overtly designed towards those applications - is a good sign that we've got a well-designed, flexible architecture.

Basically, it's the eating-your-own-dogfood thing - if it's good enough to build our applications in, using just the public API, it'll be good enough for third-party developers too.

**Comment by kaedroho** [comment]
> Being incredibly useful is not reason enough - django.contrib.admin is also incredibly useful, but that hasn't been promoted to Django core :-)

Moving ``RoutablePageMixin`` into ``wagtailcore`` is quite a different thing to moving ``django.contrib.admin`` into ``django.core``.

``RoutablePageMixin`` is a single class in its own module. We have other features in ``wagtailcore`` that could be in a separate module, but they're not because it makes more sense for them to be part of ``wagtailcore``.

I think the same for ``RoutablePageMixin``. And it doesn't need to be put in models.py as it isn't a concrete model.
